homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,223 @@
+        return {
+            "identifiers": {(AGENT_DOMAIN, self._uniqueid)},
+            "name": self._name,
+            "manufacturer": "-",
+            "model": "Camera",
+            "sw_version": self.device.client.version,
+        }
+
+    @property
+    def should_poll(self) -> bool:
+        """Update the state periodically."""
+        return True
+
+    async def async_update(self):
+        """Update our state from the Agent API."""
+        await self.device.update()
+
+    @property
+    def is_recording(self) -> bool:
+        """Return whether the monitor is recording."""
+        return self.device.recording
+
+    @property
+    def is_alerted(self) -> bool:
+        """Return whether the monitor has alerted."""
+        return self.device.alerted
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.device.client.is_available
+
+    @property
+    def connected(self) -> bool:
+        """Return True if entity is connected."""
+        return self.device.connected
+
+    @property
+    def enabled(self) -> bool:
+        """Return True if entity is enabled."""
+        return self.device.online
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return self._enabled_default
+
+    @property
+    def supported_features(self) -> int:
+        """Return supported features."""
+        return SUPPORT_ON_OFF | SUPPORT_STREAM
+
+    async def stream_source(self) -> str:
+        """Return the mp4 stream source."""
+        return self._stream_url
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this agent object."""
+        return f"{self._uniqueid}"
+
+    async def async_enable_alerts(self):
+        """Enable alerts."""
+        await self.device.alerts_on()
+
+    async def async_disable_alerts(self):
+        """Disable alerts."""
+        await self.device.alerts_off()
+
+    async def async_start_recording(self):
+        """Start recording."""
+        await self.device.record()
+
+    async def async_stop_recording(self):
+        """Stop recording."""
+        await self.device.record_stop()
+
+    async def async_turn_on(self):
+        """Enable the camera."""
+        await self.device.enable()
+
+    async def async_toggle(self):
+        """Enable/disable the camera."""
+        if self.device.online:
+            await self.device.disable()
+        else:
+            await self.device.enable()
+
+    async def async_snapshot(self):
+        """Take a snapshot."""
+        await self.device.snapshot()
+
+    async def async_turn_off(self):
+        """Disable the camera."""
+        await self.device.disable()
+
+    # Other Entity method overrides
+
+    async def async_on_demand_update(self):
+        """Update state."""
+        self.async_schedule_update_ha_state(True)
```

Comment:
```suggestion
        self.async_write_ha_state()
```

You should be able to directly write state here